sdk/azidentity/syncer.go

```diff
@@ -41,21 +46,24 @@ func newSyncer(name, tenant string, additionalTenants []string, reqToken, silent
 
 // GetToken ensures that only one goroutine authenticates at a time
 func (s *syncer) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	at := azcore.AccessToken{}
 	if len(opts.Scopes) == 0 {
-		return azcore.AccessToken{}, errors.New(s.name + ".GetToken() requires at least one scope")
+		return at, errors.New(s.name + ".GetToken() requires at least one scope")
 	}
 	// we don't resolve the tenant for managed identities because they can acquire tokens only from their home tenants
 	if s.name != credNameManagedIdentity {
 		tenant, err := s.resolveTenant(opts.TenantID)
 		if err != nil {
-			return azcore.AccessToken{}, err
+			return at, err
 		}
 		opts.TenantID = tenant
 	}
-	at, err := s.silent(ctx, opts)
-	if err != nil {
+	var err error
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.silent == nil {
```

Comment:
This is the magic right?  Since CLI and OBO credentials don't populate `s.silent` they only attempt once.

Reply:
Right. Previously, `s.silent` and `s.reqToken` were the same func for these credentials, so when `silent` returned an error like "you need to log in to the CLI", we'd waste time doing the same thing again.